### PR TITLE
175 Onboarding Scope 2 Indicator

### PR DESCRIPTION
**Pull request by drelsabrouty**
to overcome the recursion limit error, I increased the limit manually

### CODE REVIEW

File: functions/processing_enriched_datamodel.py [hunk 1]
```diff
@@ -4,6 +4,10 @@
 from functions.spark_session import create_spark_session
 from functions.dataframe_helpers import *
 from pyspark.sql.functions import col, substring
+import sys
+ 
+sys.setrecursionlimit(10**6)
```
Comment (SKruthoff) [inline functions/processing_enriched_datamodel.py:9]: Two Questions:
- Do we need 1 million?
- Can we add this to the specific part of the if condition for that procressing function?

File: functions/processing_enriched_datamodel.py [hunk 2]
```diff
@@ -409,6 +413,81 @@ def generate_table(table_name: str) -> None:
...
+        input_data_w_co2.rename_columns({"Amount":"input_amount", "Activity_UUID_Product_UUID": "activity_uuid_product_uuid", "Input_Activity_UUID_Product_UUID": "input_activity_uuid_product_uuid"})
+
+        ## PREPPING
+        joined_records = input_data_w_co2.custom_join(scope_2_emissions, custom_on=["input_activity_name", "input_reference_product_name", "input_geography"], custom_how="inner")
+        scope_2_indicator_output = joined_records.custom_select(["activity_uuid_product_uuid", "input_activity_uuid_product_uuid", "input_amount", "input_scope_2_emission"])
+
+        ## CALCULATION
+        scope_2_indicator_output.data = scope_2_indicator_output.data.withColumn("total_scope_2_emission_per_input_amount", F.col("input_amount") * F.col("input_scope_2_emission"))
+        windowSpec = Window.partitionBy("activity_uuid_product_uuid")
+        scope_2_indicator_output.data = scope_2_indicator_output.data.withColumn("total_scope_2_emission_per_activity_uuid_product_uuid", F.sum("total_scope_2_emission_per_input_amount").over(windowSpec))
```
Comment (SKruthoff) [inline functions/processing_enriched_datamodel.py:465]: Here we can use a group by much easier, but I know that the custom group by is in another PR.

File: functions/processing_enriched_datamodel.py [hunk 3]
```diff
@@ -409,6 +413,81 @@ def generate_table(table_name: str) -> None:
...
+
+        ## PREPPING
+        covered_scope_2_output = filtered_ei_record_info.custom_join(scope_2_indicator_output,custom_on="activity_uuid_product_uuid", custom_how="left").custom_select(["activity_uuid_product_uuid", "activity_name", "reference_product_name", "co2_footprint", "geography", "total_scope_2_emission_per_activity_uuid_product_uuid"])
```
Comment (SKruthoff) [inline functions/processing_enriched_datamodel.py:468]: Do we not need to make these rows unique? I could imagine that there are two or more rows for one activity uuid product uuid that have the same sum of emissions due to the sum over the window.

**Review by SKruthoff — CHANGES_REQUESTED** [review]
In general the PR looks good. I would think it is smarter to first join these two PRs (  ) because they have some functionality that might benefit you when aggregating and summing over different groups for the emissions. 

Other than that it looks good!

File: functions/processing_enriched_datamodel.py (hunk 2 above)
Comment (drelsabrouty) [reply functions/processing_enriched_datamodel.py:465]: yes agreed

File: functions/processing_enriched_datamodel.py (hunk 1 above)
Comment (drelsabrouty) [reply functions/processing_enriched_datamodel.py:9]: fixed

**Pull request merged**